tests/test_peer_heap.py

```diff
@@ -107,6 +107,64 @@ def test_update(peer_heap, peers):
     assert peer_heap.peek_peer() == p
 
 
+def test_remove(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    n = len(peers)
+    for _ in six.moves.range(n):
+        p = peer_heap.peers[random.randint(0, peer_heap.size() - 1)]
+        assert p is peer_heap.remove_peer(p)
+        verify(peer_heap, 0)
+        verify_peer_not_in_heap(peer_heap, p)
+
+
+def verify_peer_not_in_heap(peer_heap, p):
+    for peer in peer_heap.peers:
+        assert peer is not p
+
+
+def test_remove_duplicate(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    n = random.randint(0, len(peers))
+
+    for _ in six.moves.range(n):
+        p = peer_heap.peers[random.randint(0, peer_heap.size() - 1)]
+        assert p is peer_heap.remove_peer(p)
+        assert peer_heap.remove_peer(p) is None
+        verify(peer_heap, 0)
+        verify_peer_not_in_heap(peer_heap, p)
+
+
+def test_remove_from_empty_heap():
+    heap = PeerHeap()
+    heap.remove_peer(mock_peer())
+
+
+def test_remove_none_exist(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    peer_heap.remove_peer(peers[0])
+
+
+def test_remove_mismatch(peer_heap, peers):
+    for peer in peers:
+        peer_heap.push_peer(peer)
+        verify(peer_heap, 0)
+
+    # create a fake peer with duplicated index.
+    fake_peer = mock_peer()
+    fake_peer.index = 1
+    with pytest.raises(AssertionError):
+        assert fake_peer is peer_heap.remove_peer(fake_peer)
```

Comment:
What's this? This doesn't help the test. In fact it introduces the possibility that remove_peer behaves incorrectly and this test still passes.

This is not what I meant when I said you need to test that the correct assertion error was raised. You need to assert that the `AssertionError` was thrown with `pytest.raises` and then check that the error message matches what you expect it to be. See https://pytest.org/latest/assert.html#assertions-about-expected-exceptions